crates/rover-client/src/query/config/whoami.rs

```diff
@@ -43,6 +45,19 @@ pub fn run(
     get_identity_from_response_data(response_data, client.credential.origin.clone())
 }
 
+/// Get info from the registry about an API key, i.e. the name/id of the
+/// user/graph and what kind of key it is (GRAPH/USER/Other)
+/// Also prints a loading message with a progress spinner.
+#[cfg(feature = "spinners")]
+pub fn run_with_message<M: Display>(
+    variables: who_am_i_query::Variables,
+    message: M,
+    client: &StudioClient,
+) -> Result<RegistryIdentity, RoverClientError> {
+    let response_data = client.post_with_message::<WhoAmIQuery, M>(variables, message)?;
```

Comment:
I think it feels weird to me that this lives in `rover-client`. It's breaking the convention we've setup so far to let `rover` handle all output

Reply:
yeah... i tend to agree. i wonder if there's some way we can integrate the spinner into our `Command` structure instead of putting it into `rover-client`... the thing is that i was trying to get it to a point where we could just easily have it for _all_ client requests, but having that in `rover` likely requires a bit bigger refactor. i've let this sit long enough i think for now i'll just close it and we can refer back to this spike if we ever decide we need something like this.